src/components/overlay/index.js

```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Image} from 'react-native';
+import {Colors} from '../../style';
+import {PureBaseComponent} from '../../commons';
+import View from '../view';
+
+
+const gradientImage = require('./assets/GradientOverlay.png');
+const OVERLY_TYPES = {
+  DEFAULT: 'default',
+  TOP: 'top',
+  BOTTOM: 'bottom',
+  SOLID: 'solid'
+}
+/**
+ * @description: Overlay view with types (default, top, bottom, solid)
+ * @extends: Image
+ * @extendsLink: https://facebook.github.io/react-native/docs/image
+ */
+export default class Overlay extends PureBaseComponent {
+  static displayName = 'Overlay';
+
+  static propTypes = {
+    /** 
+    * The type of overlay to set on top of the image 
+    */
+    type: PropTypes.oneOf(_.values(OVERLY_TYPES))
+  };
+
+  static overlayTypes = OVERLY_TYPES;
+
+  getStyleByType() {
+    switch (this.props.type) {
+      case OVERLY_TYPES.TOP: 
+        return styles.top;
+      case OVERLY_TYPES.BOTTOM: 
+        return styles.bottom;
+      case OVERLY_TYPES.SOLID: 
+        return styles.solid
+      default:
+        break;
+    }
+  }
+
+  renderImage(typeStyle) {
+    const {type, style} = this.props;
+    const image = type !== OVERLY_TYPES.SOLID ? gradientImage : undefined;
+
+    return (
+      <Image
+        style={[style, styles.container, typeStyle, {width: undefined}]}
+        resizeMode={'stretch'}
+        source={image}
+      />
+    );
+  }
+
+  render() {
+    const {style, type} = this.props;
+    const width = _.get(style, 'width');
```

Comment:
You can't extract `width` or other props from style, cause you can't tell if style is a number/object/array. this potentially can return undefined.

Reply:
checking again, it actually does return undefined, cause you pass an array from the `Image` component. 

and if it works without it, it means the `onLayout` in the `Image` is redundant.

Reply:
Not really. The width is passed in the style object to the renderImage() as well